File: PostService/PostService-APIRoot/DeletePostMessage/DeletePostMessage.scala
package APIs.PostService

import Common.API.API
import Global.ServiceCenter.PostServiceCode

import io.circe.{Decoder, Encoder, Json}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import io.circe.syntax.*
import io.circe.parser.*
import Common.Serialize.CustomColumnTypes.{decodeDateTime,encodeDateTime}

import com.fasterxml.jackson.core.`type`.TypeReference
import Common.Serialize.JacksonSerializeUtils

import scala.util.Try

import org.joda.time.DateTime
import java.util.UUID


/**
 * DeletePostMessage
 * desc: 普通用户删除帖子接口
 * @param userToken: String (用户登录令牌，用于验证用户身份)
 * @param postID: String (帖子唯一标识符，指定需要删除的帖子)
 * @return result: String (操作结果，表示删除操作成功或失败的信息)
 */

case class DeletePostMessage(
  userToken: String,
  postID: String
) extends API[String](PostServiceCode)



case object DeletePostMessage{
    
  import Common.Serialize.CustomColumnTypes.{decodeDateTime,encodeDateTime}

  // Circe 默认的 Encoder 和 Decoder
  private val circeEncoder: Encoder[DeletePostMessage] = deriveEncoder
  private val circeDecoder: Decoder[DeletePostMessage] = deriveDecoder

  // Jackson 对应的 Encoder 和 Decoder
  private val jacksonEncoder: Encoder[DeletePostMessage] = Encoder.instance { currentObj =>
    Json.fromString(JacksonSerializeUtils.serialize(currentObj))
  }

  private val jacksonDecoder: Decoder[DeletePostMessage] = Decoder.instance { cursor =>
    try { Right(JacksonSerializeUtils.deserialize(cursor.value.noSpaces, new TypeReference[DeletePostMessage]() {})) } 
    catch { case e: Throwable => Left(io.circe.DecodingFailure(e.getMessage, cursor.history)) }
  }
  
  // Circe + Jackson 兜底的 Encoder
  given deletePostMessageEncoder: Encoder[DeletePostMessage] = Encoder.instance { config =>
    Try(circeEncoder(config)).getOrElse(jacksonEncoder(config))
  }

  // Circe + Jackson 兜底的 Decoder
  given deletePostMessageDecoder: Decoder[DeletePostMessage] = Decoder.instance { cursor =>
    circeDecoder.tryDecode(cursor).orElse(jacksonDecoder.tryDecode(cursor))
  }


}

File: Plugins/PostService/APIs/DeletePostMessage.ts
/**
 * DeletePostMessage
 * desc: 删除指定帖子并返回操作结果。
 * @param userToken: String (用户的会话令牌，用于验证当前用户身份和权限。)
 * @param postID: String (帖子唯一标识符，用于指定需要删除的帖子。)
 * @return result: String (操作结果字符串，表明帖子删除的成功与否。)
 */
import { TongWenMessage } from 'Plugins/TongWenAPI/TongWenMessage'



export class DeletePostMessage extends TongWenMessage {
    constructor(
        public  userToken: string,
        public  postID: string
    ) {
        super()
    }
    getAddress(): string {
        return "127.0.0.1:10012"
    }
}

